### ISSUE TITLE
Select mode="multiple" throws useLayoutEffect warning in Next.js

### ISSUE THREAD

**Issue by igordanchenko** [ant-design/ant-design#30396]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/awesome-moser-wr83y)

### Steps to reproduce
**Reproduce locally**
```
npx create-next-app nextjs-antd-select-bug
cd nextjs-antd-select-bug
npm install antd
npm run dev
```

Edit pages/index.js
```js
import {Select} from "antd";
import "antd/dist/antd.css";

export default function Page() {
  return (
    <div>
      <Select 
        mode="multiple" 
        style={{width: "300px"}}
      >
	      <Select.Option value="A">Option A</Select.Option>
	      <Select.Option value="B">Option B</Select.Option>
      </Select>
    </div>
  )
}
```

Open http://localhost:3000/
The warning is logged in the server console with every page refresh

**Reproduce on Codesandbox**
1. Open https://codesandbox.io/s/awesome-moser-wr83y
2. Hit Refresh button in the Browser panel
3. The warning is logged in the Terminal log

### What is expected?
No exceptions or warnings are expected.

### What is actually happening?
The following warning is logged upon every single page refresh.

```
Warning: useLayoutEffect does nothing on the server, because its effect cannot be encoded into the server renderer's output format. This will lead to a mismatch between the initial, non-hydrated UI and the intended UI. Toavoid this, useLayoutEffect should only be used in components that renderexclusively on the client. See https://reactjs.org/link/uselayouteffect-ssr for common fixes.
    at Overflow (/sandbox/node_modules/rc-overflow/lib/Overflow.js:35:32)
    at SelectSelector (/sandbox/node_modules/rc-select/lib/Selector/MultipleSelector.js:36:18)
    at div
    at Selector (/sandbox/node_modules/rc-select/lib/Selector/index.js:36:35)
    at Trigger (/sandbox/node_modules/rc-trigger/lib/index.js:79:36)
    at SelectTrigger (/sandbox/node_modules/rc-select/lib/SelectTrigger.js:66:25)
    at div
    at Select (/sandbox/node_modules/rc-select/lib/generate.js:88:34)
    at Select (/sandbox/node_modules/rc-select/lib/Select.js:95:34)
    at InternalSelect (/sandbox/node_modules/antd/lib/select/index.js:55:31)
    at div
    at Page
    at App (webpack-internal:///./node_modules/next/dist/pages/_app.js:31:1)
    at AppContainer (/sandbox/node_modules/next/dist/next-server/server/render.js:28:859)
```

| Environment | Info |
|---|---|
| antd | 4.15.4 |
| React | 17.0.2 |
| System | macOS Big Sur 11.3 |
| Browser | Safari 14.1 |

**Comment by kpirumyan** [comment]
I've got the same. Even if I use mode='tags' I'm getting the warning.

**Comment by benjie** [comment]
I'm getting the same issue with the `Menu` component also triggering this `rc-overflow` behaviour:

```
Warning: useLayoutEffect does nothing on the server, because its effect cannot be encoded into the server renderer's output format. This will lead to a mismatch between the initial, non-hydrated UI and the intended UI. To avoid this, useLayoutEffect should only be used in components that render exclusively on the client. See https://reactjs.org/link/uselayouteffect-ssr for common fixes.
    at Overflow (/home/benjie/Dev/graphile/starter/node_modules/rc-overflow/lib/Overflow.js:42:32)
    at InheritableContextProvider (/home/benjie/Dev/graphile/starter/node_modules/rc-menu/lib/context/MenuContext.js:39:23)
    at Menu (/home/benjie/Dev/graphile/starter/node_modules/rc-menu/lib/Menu.js:76:32)
    at InternalMenu (/home/benjie/Dev/graphile/starter/node_modules/antd/lib/menu/index.js:71:37)
    at Menu (/home/benjie/Dev/graphile/starter/node_modules/antd/lib/menu/index.js:162:37)
    at div
    at aside
    at Sider (/home/benjie/Dev/graphile/starter/node_modules/antd/lib/layout/Sider.js:74:31)
    at section
    at BasicLayout (/home/benjie/Dev/graphile/starter/node_modules/antd/lib/layout/layout.js:93:34)
    at Layout (/home/benjie/Dev/graphile/starter/node_modules/antd/lib/layout/layout.js:61:37)
    at main
    at Basic (/home/benjie/Dev/graphile/starter/node_modules/antd/lib/layout/layout.js:78:25)
    at Content (/home/benjie/Dev/graphile/starter/node_modules/antd/lib/layout/layout.js:61:37)
    at section
    at BasicLayout (/home/benjie/Dev/graphile/starter/node_modules/antd/lib/layout/layout.js:93:34)
    at Layout (/home/benjie/Dev/graphile/starter/node_modules/antd/lib/layout/layout.js:61:37)
    at SharedLayout (webpack-internal:///../../components/dist/SharedLayout.js:57:20)
    at SettingsLayout (webpack-internal:///../../components/dist/SettingsLayout.js:47:20)
    at Settings_Profile (webpack-internal:///./pages/settings/index.tsx:48:62)
    at ApolloProvider (/home/benjie/Dev/graphile/starter/node_modules/@apollo/client/react/context/context.cjs.js:32:21)
    at MyApp (webpack-internal:///./pages/_app.tsx:49:1)
    at WithApollo(MyApp) (/home/benjie/Dev/graphile/starter/node_modules/next-with-apollo/lib/withApollo.js:84:29)
    at AppContainer (/home/benjie/Dev/graphile/starter/node_modules/next/dist/next-server/server/render.js:28:952)
```

**Comment by HADB** [comment]
+1 Menu also has this problem.

```
Warning: useLayoutEffect does nothing on the server, because its effect cannot be encoded into the server renderer's output format. This will lead to a mismatch between the initial, non-hydrated UI and the intended UI. To avoid this, useLayoutEffect should only be used in components that render exclusively on the client. See https://reactjs.org/link/uselayouteffect-ssr for common fixes.
    at Overflow (****/node_modules/rc-overflow/lib/Overflow.js:42:32)
    at InheritableContextProvider (****/node_modules/rc-menu/lib/context/MenuContext.js:39:23)
    at Menu (****/node_modules/rc-menu/lib/Menu.js:76:32)
    at InternalMenu (****/node_modules/antd/lib/menu/index.js:71:37)
    at Menu (****/node_modules/antd/lib/menu/index.js:162:37)
    at div
    at aside
    at Sider (****/node_modules/antd/lib/layout/Sider.js:74:31)
    at section
    at BasicLayout (****/node_modules/antd/lib/layout/layout.js:93:34)
    at Layout (****/node_modules/antd/lib/layout/layout.js:61:37)
    at MainLayout (webpack-internal:///./src/components/main-layout.tsx:24:54)
    at Home
    at App (webpack-internal:///./src/pages/_app.tsx:33:3)
    at AppContainer (****/node_modules/next/dist/next-server/server/render.js:28:861)
```

**Comment by theguybyte** [comment]
Hi everyone. Any update about this issue in 4.16.0?

Thanks!

**Comment by devuxer** [comment]
Can verify, downgraded to 4.15.2 for now.

**Comment by afc163** [comment]
```
npm test
```

<img width="1262" alt="图片" src="https://user-images.githubusercontent.com/507615/120186478-22aa0880-c246-11eb-971d-6b6f0ca40158.png">

**Comment by RiceWorld** [comment]
+1 Tree component same!

**Comment by tsusini** [comment]
This happen also with  `Menu` and `SubMenu` Component !

`Warning: useLayoutEffect does nothing on the server, because its effect cannot be encoded into the server renderer's output format. This will lead to a mismatch between the initial, non-hydrated UI and the intended UI. To avoid this, useLayoutEffect should only be used in components that render exclusively on the client. See https://reactjs.org/link/uselayouteffect-ssr for common fixes.
    at Overflow (**/node_modules/rc-overflow/lib/Overflow.js:42:32)
    at InheritableContextProvider (**/node_modules/rc-menu/lib/context/MenuContext.js:39:23)
    at Menu (**/node_modules/rc-menu/lib/Menu.js:76:32)
    at InternalMenu (**/node_modules/antd/lib/menu/index.js:73:37)
    at Menu (**/node_modules/antd/lib/menu/index.js:165:37)`

**Comment by yoyo837** [comment]
Same as https://github.com/react-component/overflow/issues/6

**Comment by samora17** [comment]
+1 Menu also has this problem,
"antd": "4.16.3", "react": "17.0.2", "next": "11.0.0".

`Warning: useLayoutEffect does nothing on the server, because its effect cannot be encoded into the server renderer's output format. This will lead to a mismatch between the initial, non-hydrated UI and the intended UI. To avoid this, useLayoutEffect should only be used in components that render exclusively on the client. See https://reactjs.org/link/uselayouteffect-ssr for common fixes.
    at Overflow (.../node_modules/rc-overflow/lib/Overflow.js:42:32)
    at InheritableContextProvider (.../node_modules/rc-menu/lib/context/MenuContext.js:39:23)
    at Menu (.../node_modules/rc-menu/lib/Menu.js:76:32)
    at InternalMenu (.../node_modules/antd/lib/menu/index.js:73:37)
    at Menu (.../node_modules/antd/lib/menu/index.js:165:37)
    at Nav
    at Layout (webpack-internal:///./components/Layout.js:15:3)
    at MyApp (webpack-internal:///./pages/_app.js:32:3)
    at AppContainer (.../node_modules/next/dist/next-server/server/render.js:27:952)`

**Comment by xuerzong** [comment]
I have the same problem !!! This problem occurs when I upgrade `antd` from `^4.0.0` to `^4.0.5`.

"antd":  `^4.5.0`
"next": `^9.2.1`
"react": `^16.12.0`

```bash
Warning: useLayoutEffect does nothing on the server, because its effect cannot be encoded into the server renderer's output format. This will lead to a mismatch between the initial, non-hydrated UI and the intended UI. To avoid this, useLayoutEffect should only be used in components that render exclusively on the client. See https://fb.me/react-uselayouteffect-ssr for common fixes.
    in Overflow
    in Context.Provider
    in Context.Provider
    in InheritableContextProvider
    in Context.Provider
    in Menu
    in Context.Provider
    in Context.Consumer
    in InternalMenu
    in Context.Consumer
    in Menu
    in div
    in aside
    in Context.Provider
    in Sider
    in section
    in Context.Provider
    in BasicLayout
    in Layout
    in section
    in Context.Provider
    in BasicLayout
    in Layout
    in Layout
    in Context.Provider
    in Provider
    in Context.Provider
    in ProviderChildren
    in Context.Consumer
    in LocaleReceiver
    in ConfigProvider
    in Fragment
    in MyApp
    in withRedux(MyApp)
    in Context.Provider
    in Context.Provider
    in ae
    in Unknown
    in Context.Provider
    in Context.Provider
    in Context.Provider
    in AppContainer

```

**Comment by devuxer** [comment]
I'm going to be stuck on 4.15.2 forever if this doesn't get fixed. @afc163, any plans to fix?

**Comment by flex-junwoo** [comment]
hi, any update here?

**Comment by allancomlogik** [comment]
Same problem any fix?

**Comment by PauloRodrigues33** [comment]
Same problem, waiting fix.

**Comment by aprisniak** [comment]
Same problem, waiting fix.

**Comment by Wave-SYJ** [comment]
Have the same problem. 
"antd": "4.16.11", "react": "17.0.2", "next": "11.1.0".
Wanting fix.

**Comment by as5908** [comment]
Have the same issue with even 4.10.0, tried downgrading to 4.15.2, 4.12.0 as well. Ended up downgrading to 4.8.0
react:16.13.1 next:10.0.9

**Comment by born2bewild** [comment]
> Have the same issue with even 4.10.0, tried downgrading to 4.15.2, 4.12.0 as well. Ended up downgrading to 4.8.0
> react:16.13.1 next:10.0.9

It works fine, for me, on 4.15.2, w/o useLayoutEffect warning.

**Comment by as5908** [comment]
> > Have the same issue with even 4.10.0, tried downgrading to 4.15.2, 4.12.0 as well. Ended up downgrading to 4.8.0
> > react:16.13.1 next:10.0.9
> 
> It works fine, for me, on 4.15.2, w/o useLayoutEffect warning.

Just to confirm - Worked for Select component with mode='multiple' ? What version is your react and next ?

**Comment by hojas** [comment]
Same problem, waiting fix.

**Comment by Mihai-github** [comment]
+1

**Comment by McDaddy** [comment]
same problem.... any update for this ?

**Comment by antaiSlasify** [comment]
same problem.... any update for this ?

**Comment by CMatheus04** [comment]
same problem, waiting fix.

**Comment by aacgn** [comment]
same problem. waiting fix too

**Comment by nodkz** [comment]
Almost 3 months ago was opened PR with fix https://github.com/react-component/overflow/pull/16
@afc163 / @zombieJ code is trivial but it's really difficult to add tests for such case. Maybe you accept PR without tests?

**Comment by timnlupo** [comment]
Same problem waiting on fix

**Comment by theguybyte** [comment]
Waiting fix

**Comment by muyu2019** [comment]
Waiting fix

**Comment by 06kellyjac** [comment]
Can people please stop commenting with non-updates. It's quite frustrating to check for progress just to see another waiting comment.

Please subscribe and thumbs up the original post, and then one person can bump the issue every so often, not within a couple hours of each other.

**Comment by hamidgarousi** [comment]
hi everybody
i find a solution for this problem 
I use this line in _document.tsx and useLayoutEffect warning resolved

```
import React from 'react';
React.useEffectLayout = React.useEffect
```

my _document.tsx file :


```
// eslint-disable-next-line @next/next/no-document-import-in-page
import Document, {
  Html,
  DocumentContext,
  Head,
  Main,
  NextScript,
} from "next/document";

import React from "react";

React.useLayoutEffect = React.useEffect;

class MyDocument extends Document {
  static async getInitialProps(ctx: DocumentContext) {
    const initialProps = await Document.getInitialProps(ctx);
    return { ...initialProps };
  }

  render() {
    return (
      <Html>
        <Head></Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    );
  }
}

export default MyDocument;
```

**Comment by lucidsinc** [comment]
I was using the menu component inside nextjs when i came through this error, since my menu doesn't need to change after the initial render, i basically wrapped my menu component inside a useEffect Hook :) Worked like a charm

`{
  useEffect(() => {
    // ...
  });
}`

**Comment by zaistev** [comment]
I just added this beautiful code in the file that uses 

```
import { Layout, Menu, Spin } from "antd";
```

```
// Ugly hotifx
// https://github.com/ant-design/ant-design/issues/30396
React['useEffectLayout'] = React.useEffect;

```

**Comment by Mihai-github** [comment]
For me, nothing from the above solution didn't work, I've tried:
* downgrade the version of antd
* use React.useLayoutEffect = React.useEffect; in _document.tsx

Maybe some have another solution because is kinda annoying to get this large warning message In the console... making debugging harder also.

**Comment by Mihai-github** [comment]
> hi everybody i find a solution for this problem I use this line in _document.tsx and useLayoutEffect warning resolved
> 
> ```
> import React from 'react';
> React.useEffectLayout = React.useEffect
> ```
> 
> my _document.tsx file :
> 
> ```
> // eslint-disable-next-line @next/next/no-document-import-in-page
> import Document, {
>   Html,
>   DocumentContext,
>   Head,
>   Main,
>   NextScript,
> } from "next/document";
> 
> import React from "react";
> 
> React.useLayoutEffect = React.useEffect;
> 
> class MyDocument extends Document {
>   static async getInitialProps(ctx: DocumentContext) {
>     const initialProps = await Document.getInitialProps(ctx);
>     return { ...initialProps };
>   }
> 
>   render() {
>     return (
>       <Html>
>         <Head></Head>
>         <body>
>           <Main />
>           <NextScript />
>         </body>
>       </Html>
>     );
>   }
> }
> 
> export default MyDocument;
> ```

For me this one worked

**Comment by anonymous-coder** [comment]
@Mihai-github worked like a charm, ty

**Comment by ChuckJonas** [comment]
@Mihai-github seems like there could be some negative impacts from this solution?  Although I'm not sure exactly...

**Comment by Wave-SYJ** [comment]
> hi everybody i find a solution for this problem I use this line in _document.tsx and useLayoutEffect warning resolved
> 
> ```
> import React from 'react';
> React.useEffectLayout = React.useEffect
> ```
> 
> my _document.tsx file :
> 
> ```
> // eslint-disable-next-line @next/next/no-document-import-in-page
> import Document, {
>   Html,
>   DocumentContext,
>   Head,
>   Main,
>   NextScript,
> } from "next/document";
> 
> import React from "react";
> 
> React.useLayoutEffect = React.useEffect;
> 
> class MyDocument extends Document {
>   static async getInitialProps(ctx: DocumentContext) {
>     const initialProps = await Document.getInitialProps(ctx);
>     return { ...initialProps };
>   }
> 
>   render() {
>     return (
>       <Html>
>         <Head></Head>
>         <body>
>           <Main />
>           <NextScript />
>         </body>
>       </Html>
>     );
>   }
> }
> 
> export default MyDocument;
> ```

This works for me. Thanks a lot!

**Comment by atom-shanghai** [comment]
Warning: useLayoutEffect does nothing on the server, because its effect cannot be encoded into the server renderer's output format. This will lead to a mismatch between the initial, non-hydrated UI and the intended UI. To avoid this, useLayoutEffect should only be used in components that render exclusively on the client. See https://reactjs.org/link/uselayouteffect-ssr for common fixes.

**Comment by originalix** [comment]
next: 12.0.4, react: 17.0.2, antd: 4.17.0.
I have the same problem when I use Select component.

**Comment by moshest** [comment]
Should be fixed on the library as follow:

```ts
const useIsomorphicLayoutEffect = typeof window !== 'undefined' ? useLayoutEffect : useEffect
```

Credit: https://github.com/react-component/overflow/issues/6#issuecomment-819215239

**Comment by MartinSchmidt123** [comment]
The tree component is still affected by this in antd 4.18.2.

I use nextjs and my workaround is to dynamically load the component containing the tree and disabling ssr for it.
Instead of 
```
import { Tree } from 'antd';
<Tree [..] />
```
I am using
```
import dynamic from 'next/dynamic';
const MyTree = dynamic(() => import('./MyTree'), {ssr: false});

[..]

<MyTree [..] />
```
with a file `MyTree.js` as 
```
import { Tree } from 'antd';
const MyTree = () => {
 <Tree [..] />
}
export default MyTree
```

**Comment by honam867** [comment]
> import dynamic from 'next/dynamic';

work perfectly with   "antd": "^4.18.3",  "next": "12.0.4" and "react": "17.0.2",

**Comment by originalix** [comment]
> The tree component is still affected by this in antd 4.18.2.
> 
> I use nextjs and my workaround is to dynamically load the component containing the tree and disabling ssr for it. Instead of
> 
> ```
> import { Tree } from 'antd';
> <Tree [..] />
> ```
> 
> I am using
> 
> ```
> import dynamic from 'next/dynamic';
> const MyTree = dynamic(() => import('./MyTree'), {ssr: false});
> 
> [..]
> 
> <MyTree [..] />
> ```
> 
> with a file `MyTree.js` as
> 
> ```
> import { Tree } from 'antd';
> const MyTree = () => {
>  <Tree [..] />
> }
> export default MyTree
> ```

thank you, that's worked for me! ❤️

**Comment by hallo-yurik** [comment]
I updated the version from "4.16.13" to "4.18.4" and now I get a lot of the following warnings:
```
Warning: useLayoutEffect does nothing on the server, because its effect cannot be encoded into the server renderer's output format. This will lead to a mismatch between the initial, non-hydrated UI and the intended UI. To avoid this, useLayoutEffect should only be used in components that render exclusively on the client. See https://reactjs.org/link/uselayouteffect-ssr for common fixes.
    at Ellipsis
    at span
    at Typography
    ...
```

After a little debugging, I found that the latest version of antd uses useLayoutEffect for the "ellipsis" property. This is a tragedy for me, because I use this component in the whole project. But the saddest thing is that I never used the "ellipsis" property. And now, every time I refresh the page, I get hundreds of lines in the console about the warning, which I can not easily fix, and among all this text to see the real errors that may occur during development.

Hopefully, the developers will be able to follow the recommendations described at the link "https://reactjs.org/link/uselayouteffect-ssr", so that this does not happen at least with the component that is used so often.

**Comment by MarkLyck** [comment]
It's February 2022, and this still isn't fixed?

@afc163 please add this one line of code to solve this 🙏

`const useIsomorphicLayoutEffect = typeof window !== 'undefined' ? useLayoutEffect : useEffect`

**Comment by By-Ha** [comment]
> It's February 2022, and this still isn't fixed?
> 
> @afc163 please add this one line of code to solve this 🙏
> 
> `const useIsomorphicLayoutEffect = typeof window !== 'undefined' ? useLayoutEffect : useEffect`

It has been fixed but not yet merged. #33818

**Comment by 06kellyjac** [comment]
#33818 merged now

**Comment by emintokluu** [comment]
I updated to version 4.18.7 but I still have the same problem. Menu in Next js

`Warning: useLayoutEffect does nothing on the server, because its effect cannot be encoded into the server renderer's output format. This will lead to a mismatch between the initial, non-hydrated UI and the intended UI. To avoid this, useLayoutEffect should only be used in components that render exclusively on the client. See https://reactjs.org/link/uselayouteffect-ssr for common fixes.
    at Overflow (C:\dev\semerkand\kurban\node_modules\rc-overflow\lib\Overflow.js:42:32)
    at InheritableContextProvider (C:\dev\semerkand\kurban\node_modules\rc-menu\lib\context\MenuContext.js:40:23)
    at Menu (C:\dev\semerkand\kurban\node_modules\rc-menu\lib\Menu.js:80:32)
    at InternalMenu (C:\dev\semerkand\kurban\node_modules\antd\lib\menu\index.js:77:37)
    at Menu (C:\dev\semerkand\kurban\node_modules\antd\lib\menu\index.js:176:37)
    at div
    at div
    at div
    at header
    at Basic (C:\dev\semerkand\kurban\node_modules\antd\lib\layout\layout.js:78:25)
    at Adapter (C:\dev\semerkand\kurban\node_modules\antd\lib\layout\layout.js:61:37)
    at Navbar
    at section
    at BasicLayout (C:\dev\semerkand\kurban\node_modules\antd\lib\layout\layout.js:93:34)
    at Adapter (C:\dev\semerkand\kurban\node_modules\antd\lib\layout\layout.js:61:37)
    at ContentComp (webpack-internal:///./components/content.tsx:17:32)
    at Index (webpack-internal:///./pages/index.tsx:42:26)
    at MyApp (webpack-internal:///./pages/_app.tsx:21:18)
    at StyleRegistry (C:\dev\semerkand\kurban\node_modules\styled-jsx\dist\index\index.js:671:34)
    at AppContainer (C:\dev\semerkand\kurban\node_modules\next\dist\server\render.js:394:29)
    at AppContainerWithIsomorphicFiberStructure (C:\dev\semerkand\kurban\node_modules\next\dist\server\render.js:424:57)
    at div
    at Body (C:\dev\semerkand\kurban\node_modules\next\dist\server\render.js:701:21)`

**Comment by rennehir** [comment]
For me, upgrading `antd` to version 4.18.7 seems to be solving the problem 🚀

**Comment by yoyo837** [comment]
> I updated to version 4.18.7 but I still have the same problem. Menu in Next js
> 
> ```
>     at Overflow (C:\dev\semerkand\kurban\node_modules\rc-overflow\lib\Overflow.js:42:32)
>     at InheritableContextProvider (C:\dev\semerkand\kurban\node_modules\rc-menu\lib\context\MenuContext.js:40:23)
>     at Menu (C:\dev\semerkand\kurban\node_modules\rc-menu\lib\Menu.js:80:32)
>     at InternalMenu (C:\dev\semerkand\kurban\node_modules\antd\lib\menu\index.js:77:37)
>     at Menu (C:\dev\semerkand\kurban\node_modules\antd\lib\menu\index.js:176:37)
>     at div
>     at div
>     at div
>     at header
>     at Basic (C:\dev\semerkand\kurban\node_modules\antd\lib\layout\layout.js:78:25)
>     at Adapter (C:\dev\semerkand\kurban\node_modules\antd\lib\layout\layout.js:61:37)
>     at Navbar
>     at section
>     at BasicLayout (C:\dev\semerkand\kurban\node_modules\antd\lib\layout\layout.js:93:34)
>     at Adapter (C:\dev\semerkand\kurban\node_modules\antd\lib\layout\layout.js:61:37)
>     at ContentComp (webpack-internal:///./components/content.tsx:17:32)
>     at Index (webpack-internal:///./pages/index.tsx:42:26)
>     at MyApp (webpack-internal:///./pages/_app.tsx:21:18)
>     at StyleRegistry (C:\dev\semerkand\kurban\node_modules\styled-jsx\dist\index\index.js:671:34)
>     at AppContainer (C:\dev\semerkand\kurban\node_modules\next\dist\server\render.js:394:29)
>     at AppContainerWithIsomorphicFiberStructure (C:\dev\semerkand\kurban\node_modules\next\dist\server\render.js:424:57)
>     at div
>     at Body (C:\dev\semerkand\kurban\node_modules\next\dist\server\render.js:701:21)```
> ```

Try to delete `node_modules` and lock file(`package-lock.json` or `yarn.lock`)then reinstall.

**Comment by yoyo837** [comment]
https://github.com/react-component/overflow/issues/6 has been merged, can we safely close this?

**Comment by bnrobinson93** [comment]
Seems fixed in 4.18.8. Thanks!

**Comment by linchengzzz** [comment]
`Menu` component in Next.js has same issue

**Comment by AntoineGerv** [comment]
> `Menu` component in Next.js has same issue

I had the same problem with the antd menu, to fix it, just add last 'rc-overflow' in your package.json : 

`"rc-overflow": "^1.2.3",`

**Comment by yoyo837** [comment]
Fixed. please file a new issue if anyone still having this problems.

**Comment by joebnb** [comment]
Table also have same problem

**Comment by xavimodo** [comment]
Any solution about this? Table also have the same problem

**Comment by yoyo837** [comment]
> Try to delete `node_modules` and lock file(`package-lock.json` or `yarn.lock`) then reinstall.

Try this way please.

**Comment by xavimodo** [comment]
> > Try to delete `node_modules` and lock file(`package-lock.json` or `yarn.lock`) then reinstall.
> 
> Try this way please.

Did it but still happening. 

This is my package.json: 

```json

{
  "name": "project-name",
  "version": "0.1.0",
  "private": true,
  "prettier": "@modostudio/prettier-config",
  "scripts": {
    "dev:local": "cross-env ENV=local ./env/envscript.sh && next dev",
    "dev:staging": "cross-env ENV=staging ./env/envscript.sh && next dev",
    "dev:production": "cross-env ENV=production ./env/envscript.sh && next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint",
    "test": "jest src --coverage",
    "prepare": "husky install"
  },
  "dependencies": {
    "@svgr/webpack": "6.3.1",
    "antd": "4.21.6",
    "antd-country-phone-input": "4.3.3",
    "axios": "0.27.2",
    "jwt-decode": "3.1.2",
    "next": "12.2.2",
    "next-i18next": "11.0.0",
    "next-plugin-antd-less": "1.8.0",
    "nookies": "2.5.2",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "sass": "1.53.0",
    "world_countries_lists": "2.6.0"
  },
  "devDependencies": {
    "@modostudio/eslint-config-typescript": "1.0.2",
    "@modostudio/prettier-config": "1.0.1",
    "@testing-library/react": "13.3.0",
    "@types/jest": "28.1.6",
    "@types/node": "18.0.3",
    "@types/react": "18.0.15",
    "@types/react-dom": "18.0.6",
    "@types/sinon": "10.0.13",
    "babel-jest": "28.1.3",
    "babel-plugin-import": "1.13.5",
    "cross-env": "7.0.3",
    "eslint": "8.19.0",
    "eslint-config-next": "12.2.2",
    "husky": "8.0.1",
    "identity-obj-proxy": "3.0.0",
    "jest": "28.1.3",
    "jest-environment-jsdom": "28.1.3",
    "jest-watch-typeahead": "2.0.0",
    "prettier": "2.7.1",
    "sinon": "14.0.0",
    "ts-jest": "28.0.7",
    "typescript": "4.7.4"
  }
}
```

**Comment by laux98** [comment]
Same issue with using Table Composant, even if i delete and reinstall node-modules, in version 5.16.4.

**Comment by zoffyzhang** [comment]
Tree component still having this problem.

```json
"antd": "^5.25.0",
 "react": "^18.2.0",
 "@remix-run/dev": "^2.16.5",
```
